mantidproject/mantid
# Refactor ExportSampleLogsToHDF5 to use dtype function.

**Description of work.**
Cleaned up code in `ExportSampleLogsToHDF5.py` and the corresponding test file `ExportSampleLogsToHDF5Test.py` by removing many of the calls made to `isinstance` and using the newly implemented `dtype` function ( see issue #22553 ).

Exposed a new `PropertyFactory` class to the Python side which allows for `TimeSeriesProperty` objects to be easily created ( see issue #22555 ) using the `createTimeSeries` function which was also implemented.

This PR depends on issue #22553.

**Report to:** @samueljackson92  

**To test:**
Not much to test, just run the unit tests and check that they all pass.


Run `ctest -C Debug -R ExportSampleLogsToHDF5Test -V` for full information when running the tests.

Fixes #22554 . 

Does this update require release notes?


---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
